### DISCUSSION TITLE
Review #1

### DISCUSSION THREAD

**Discussion by daniel-ciocirlan** [riverwalk-software/astro-test#99] [Announcements]
I walked through the site in a bit more detail over the past couple of days - here are some things I found that I believe we should fix - most of these should be trivial 

General
- [x] pick color palette for
  - [x] page backdrop
  - [x] articles/videos/course cards
  - [x] article tags
  - [x] text
  - [x] links
  - [x] buttons
  - [x] CTA button
  - [x] curriculum sections
- [x] dark color palette: https://coolors.co/palette/001021-001625-011c2a-01222e-012832-022f37-02353b-023b3f-034144-034748
- [x] light color palette: TODO
- [x] add a minimal content margin for any width breakpoint
- [x] bug: no API key in Teachable calls
- [x] bug: I think the membership stats adds some course stats twice (probably need to exclude the bundle)

Homepage
- [x] too big spacing between hero and "latest content"
- [x] articles/videos cards
  - [x] use an :after pseudoelement with less width, rather than using border-bottom
- [x] help delimit the articles/videos/course cards with borders and/or shadows

Courses
- [x] better delimited cards (same as homepage)
- [x] make titles smaller
- [x] feature: maybe group the cards by categories?

Course pages
- [x] Maybe make the hero image less than 100vh
  - [x] suggestion: maybe keep the image at 100vw but crop the viewport of the image at maybe 50vh or a fixed pixel/rem height
- [x] make regular text smaller and more compact (smaller row spacing)
- [x] reduce row spacing in lists
- [x] make the "money back" image smaller
- [x] make the author images smaller and rounded
- [x] improve the contrast of the curriculum section
  - [x] course section, the "start" buttons etc
- [x] remove the "skills you'll learn" section from bundle pages
  - [x] or should I just add some content there?
- [x] bundle pages: improve contrast of section headers
- [x] feature: are you able to include the duration of each lesson in the curriculum?
- [x] bug: the "start" button leads to a localhost page, should change to the course website
- [x] Typelevel Rite of Passage: the "how this course works" should have everything full width (speedrun, text, video)
- [x] ZIO: there's an "Is there more to Kotlin" section which doesn't make sense, and an empty "Goal" section (probably needs corresponding content)
- [x] ZIO Rite of Passage has the content of "Advanced Kotlin"
- [x] curriculum takes a bit of time to load (~2 seconds)
  - [x] maybe add a "loading" status in that section?

Contact
- [x] clicking on "privacy policy" triggers the radio rather than the redirect

Articles
- [x] make the image an anchor that leads to the article
- [x] increase contrast of the excerpt (especially in dark mode)

Features
- [x] help me make a Black Friday home page and a banner to put on every course page - [ ] I have the

Minor
- [x] maybe generate sitemap?
- [x] jQuery is not defined

**Comment by andreilgeorgescu** [comment]
Thanks a lot for the feedback. Will and I looked over all your points and we have a few questions for clarification:

- For the courses page, are you essentially stating you want an accordion in the same style as the homepage instead of just listing out the courses?
- For the black friday point, that would be outside the scope of the MVP so I think that's something we can do right after we release the site but of course we need to have this be fully released before then to have time to add the black friday content.
- For the duration of each lesson in the curriculum, we could not find this in the API. I've emailed Teachable about their API in the past so I'll ask them to make sure but I don't think there's an automated way to do this right now.
- For the hero image, do you think it's just too big or is it more that you only want part of the image to be shown?
no API key in Teachable calls is this still an issue after using the .env key I sent?
- I don't think we can use ::after in tailwind. What is it that you're trying to use it for?
- What does bottom refer to in relation to ::after?
- Is bundle pages: "take " and unfinished task?
- Typelevel Rite of Passage: the "how this course works" should have everything full width (speedrun, text, video) So for the typelevel rite of passage you'd like a "how this course works" that is different than the one being used in the regular course pages where everything is full-screen including the text above the video? Is this something you'd like for every rite of passage course or just this one course as an exception?
- Were you able to recreate the background tint issue?
- For the Trustpilot account, I don't think it's worth adding the Trustpilot component until we have a number of reviews so that is something we can do after the MVP.
- We looked at more options for stock photos, generating images and cropping the banner like you said. We think that none of them really look good and it's fairly difficult to have images match computer science topics. Baeldung doesn't have any images and Real Python has custom made ones for each article, which is ideal, but probably not cheap unless they have somebody on retainer. It would probably be for the best to just not have images for now and see if we can find somebody to make custom images for articles going forward if that is something you're interested in since it's probably not worth it to retroactively make images. If you're ok with this then we can just make the images optional for the old ones based on a date refinement in Zod.

Also, I converted the issue into a discussion and made issues out of each actionable point.
And I added you to issues relating to missing course content.

**Comment by andreilgeorgescu** [comment]
Daniel's reply:

To your points:
> For the courses page, are you essentially stating you want an accordion in the same style as the homepage instead of just listing out the courses?

Just separate the courses by category so that they're easy to parse - no need for accordion or to make categories collapsible in any way

> For the black friday point, that would be outside the scope of the MVP so I think that's something we can do right after we release the site but of course we need to have this be fully released before then to have time to add the black friday content.

Sure!

> For the duration of each lesson in the curriculum, we could not find this in the API. I've emailed Teachable about their API in the past so I'll ask them to make sure but I don't think there's an automated way to do this right now.

Okay

> For the hero image, do you think it's just too big or is it more that you only want part of the image to be shown?

I think it's too big

> no API key in Teachable calls is this still an issue after using the .env key I sent?

Not in my recent tests. Might have been caused by browser cache when I first added the env. Ignore

> I don't think we can use ::after in tailwind. What is it that you're trying to use it for?

I was trying to mimic the design on the [original homepage](http://rockthejvm.com/), where the separator between the items is less than list item width.

> What does bottom refer to in relation to ::after?

Typo - I wrote border-bottom, sorry about the confusion

> Is bundle pages: "take " and unfinished task?

Removed

> Typelevel Rite of Passage: the "how this course works" should have everything full width (speedrun, text, video) So for the typelevel rite of passage you'd like a "how this course works" that is different than the one being used in the regular course pages where everything is full-screen including the text above the video? Is this something you'd like for every rite of passage course or just this one course as an exception?

It's for this course only, but I think this can be built out of two full-width parts, one with the speedrun and one with the text. I also don't think the Typelevel Rite of Passage content is ordered the same as the [original page](https://rockthejvm.com/p/typelevel-rite-of-passage)

> Were you able to recreate the background tint issue?

No, all good

> For the Trustpilot account, I don't think it's worth adding the Trustpilot component until we have a number of reviews so that is something we can do after the MVP.

Agree

> We looked at more options for stock photos, generating images and cropping the banner like you said. We think that none of them really look good and it's fairly difficult to have images match computer science topics. Baeldung doesn't have any images and Real Python has custom made ones for each article, which is ideal, but probably not cheap unless they have somebody on retainer. It would probably be for the best to just not have images for now and see if we can find somebody to make custom images for articles going forward if that is something you're interested in since it's probably not worth it to retroactively make images. If you're ok with this then we can just make the images optional for the old ones based on a date refinement in Zod.

I'm good with not having images - in this case, might be better to show articles as a list instead of grid? (edited)
